File: Exam_Server/src/main/java/com/exam/Service/QuestionService.java
package com.exam.Service;

import java.util.Set;

import org.springframework.stereotype.Service;

import com.exam.entity.exam.Questions;
import com.exam.entity.exam.Quiz;

@Service
public interface QuestionService {

	public Questions addQuestion(Questions questions);
	
	public Questions updateQuestion(Questions questions);
	
	public Set<Questions>getQuestions();
	
	public Questions getQuestion(Long questionId);
	
	public Set<Questions> getQuestionsOfQuiz(Quiz quiz);
	
	public void deleteQuestion(Long quesId);
	
	
}

File: Exam_Front/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http:HttpClient) { }


  public getQuestionsOfQuiz(qid:number){
   return  this._http.get(`${baseUrl}/questions/quiz/all/${qid}`)
  }

  public addQuestion(question:any){
    return this._http.post(`${baseUrl}/questions/`,question);
  }

  public deleteQuestion(quesId:number){
    return this._http.delete(`${baseUrl}/questions/${quesId}`);
  }

  public updateQuestion(question:any){
    return this._http.put(`${baseUrl}/questions/`,question);
  }

  // single question
  public getOneQuestion(quesId:number){
    return this._http.get(`${baseUrl}/questions/${quesId}`);
  }

}
